pyiceberg/catalog/__init__.py

```diff
@@ -646,6 +646,13 @@ def purge_table(self, identifier: Union[str, Identifier]) -> None:
         delete_files(io, prev_metadata_files, PREVIOUS_METADATA)
         delete_files(io, {table.metadata_location}, METADATA)
 
+    def table_exists(self, identifier: Union[str, Identifier]) -> bool:
```

Comment:
Hi @anupam-saini This looks great! In addition to this default behavior, I think there is a suggestion on #507 to override this method in [catalog/rest.py](https://github.com/apache/iceberg-python/blob/pyiceberg-0.6.x/pyiceberg/catalog/rest.py) to instead make a head request to the Rest Catalog at [/v1/{prefix}/namespaces/{namespace}/tables/{table}](https://github.com/apache/iceberg/blob/main/open-api/rest-catalog-open-api.yaml#L814) endpoint. Shall we incorporate that suggestion to use the available endpoint in the Rest Catalog?